### PR TITLE
[ELY-983] wildfly.sasl.mechanism-query-all should only influence getMechanismNames and not createSasl* methods.

### PR DESCRIPTION
**Pull request by darranl**
…

### CODE REVIEW

File: src/main/java/org/wildfly/security/sasl/plain/PlainSaslClientFactory.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
     public SaslClient createSaslClient(final String[] mechanisms, final String authorizationId, final String protocol, final String serverName, Map<String, ?> props, final CallbackHandler cbh) throws SaslException {
         Assert.checkNotNullParam("cbh", cbh);
         if (props == null) props = Collections.emptyMap();
-        if (PlainSasl.isMatched(props)) for (String mechanism : mechanisms) {
+        if (PlainSasl.isMatched(props, true)) for (String mechanism : mechanisms) {
```
Comment (honza889) [inline src/main/java/org/wildfly/security/sasl/plain/PlainSaslClientFactory.java]: Looks like there are true/false for query conversely - should not be false for create and true for getMechanismNames?

Comment (darranl) [inline src/main/java/org/wildfly/security/sasl/plain/PlainSaslClientFactory.java]: Good catch

**Comment by msve** [comment]
Downstream jira https://issues.jboss.org/browse/JBEAP-9231 has dev nack. If you're going to merge this, don't forget to change the it to ack+ please.

**Comment by dmlloyd** [comment]
I'm not sure that this won't break our mechanism selection assumptions though.

**Pull request merged**